### PR TITLE
selector_target field to set label source for clusterpodconvention matchers

### PR DESCRIPTION
**Pull request by katmutua**
Add a `SelectorTarget` field  on the `ClusterPodConvention` struct to allow users to define which labels the `ClusterPodConvention` matcher definitions consider.  The two options available would be labels on the 
1. PodIntent .`metadata` field (configured with value `PodIntent` on the `selectorTarget`)  or 
2. PodTemplateSpec `.spec.template.metadata` field (configured by  setting the default  value to `PodTemplateSpec`).

The current implementation only allows matching against labels defined in the  `PodIntent` .`spec.template.metadata` field Labels specified on the  PodIntent's  .`metadata` field are ignored and the ClusterPodConvention is not applied in this case. Given the following ClusterPodConvention
```yaml
---
apiVersion: conventions.apps.tanzu.vmware.com/v1alpha1
kind: ClusterPodConvention
metadata:
...
spec:
  selectors:
    - matchLabels:
       environment: production 
  webhook:
    clientConfig:
      service:
        name: sample-webhook
        namespace: sample-convention
```
The PodIntent specified below is currently evaluated as follows 
```yaml 
---
apiVersion: conventions.apps.tanzu.vmware.com/v1alpha1
kind: PodIntent
metadata:
  name: my-sample
  labels:
    severity: critical
    app: nginx
    environment: production # label is currently not considered and conventions are ignored
spec:
  template:
    metadata:
      labels: 
        environment: production # labels evaluated and convention applied
    spec:
      containers:
        - name: workload
          image: <image>
```

Proposed optional field **selectorTarget**  will be configured on the ClusterPodConvention `.spec.selectorTarget` field as follows.
```yaml
---
apiVersion: conventions.apps.tanzu.vmware.com/v1alpha1
kind: ClusterPodConvention
metadata:
...
spec:
  selectorTarget: "PodTemplateSpec" # < options include 'PodIntent' or 'PodTemplateSpec';  default value is set to 'PodTemplateSpec' >
  selectors:
    - matchLabels:
       environment: production 
  webhook:
    clientConfig:
      service:
        name: sample-webhook
        namespace: sample-convention
```

### CODE REVIEW

**Review by scothis — COMMENTED** [review]
We should also add a checks in the [defaulter](https://github.com/vmware-tanzu/cartographer-conventions/blob/2ba5314dc8ea3393573e73fefee6dbbff51161d8/pkg/apis/conventions/v1alpha1/clusterpodconvention_defaults.go#L33-L40) and [validator](https://github.com/vmware-tanzu/cartographer-conventions/blob/main/pkg/apis/conventions/v1alpha1/clusterpodconvention_validation.go#L58-L83)

File: pkg/apis/conventions/v1alpha1/clusterpodconvention_validation.go [hunk 1]
```diff
@@ -79,6 +79,13 @@ func (s *ClusterPodConventionSpec) Validate() validation.FieldErrors {
 		errs = errs.Also(s.Webhook.Validate().ViaField("webhook"))
 	}
 
+	if s.SelectorTarget != PodTemplateSpecLabels && s.SelectorTarget != PodIntentLabels {
+		errs = errs.Also(validation.FieldErrors{
+			field.Invalid(field.NewPath("selectorTarget"), s.SelectorTarget,
+				"Accepted selector target values are \"PodIntent\" and \"PodTemplate\". The default value is set to \"PodTemplateSpec\""),
```
Comment (kkavitha) [inline pkg/apis/conventions/v1alpha1/clusterpodconvention_validation.go:85]: **nit:** Change PodTemplate to PodTemplateSpec

File: pkg/apis/conventions/v1alpha1/clusterpodconvention_validation.go [hunk 2]
```diff
@@ -79,6 +79,13 @@ func (s *ClusterPodConventionSpec) Validate() validation.FieldErrors {
 		errs = errs.Also(s.Webhook.Validate().ViaField("webhook"))
 	}
 
+	if s.SelectorTarget != PodTemplateSpecLabels && s.SelectorTarget != PodIntentLabels {
+		errs = errs.Also(validation.FieldErrors{
+			field.Invalid(field.NewPath("selectorTarget"), s.SelectorTarget,
+				"Accepted selector target values are \"PodIntent\" and \"PodTemplateSpec\". The default value is set to \"PodTemplateSpec\""),
```
Comment (rszumlakowski) [inline pkg/apis/conventions/v1alpha1/clusterpodconvention_validation.go:85]: Nit: If you used backticks around the string then you wouldn't have to escape the quotes inside the string.  The code would be a bit easier to read.

```suggestion
				`Accepted selector target values are "PodIntent" and "PodTemplateSpec". The default value is set to "PodTemplateSpec"`),
```

**Comment by rszumlakowski** [comment]
Looks good to me.

**Pull request merged**